### ISSUE TITLE
Keyboard focus is moving on Checkbox within Nav two times while navigating using down arrow key

### ISSUE THREAD

**Issue by yangster23** [microsoft/fluentui#16982]
### Environment Information

- **Package version(s)**: 7.64
- **Browser and OS versions**: Windows Version: Win10, Chrome Version: 87.0.4280.88 (Official Build) (64-bit)

### Describe the issue:

Keyboard focus is moving on Checkbox within Nav two times while navigating using down arrow key.

Note that we are intentionally calling upon the onRenderLink rather than linkAs prop. This allows the user to click on the checkboxes within the Nav. 

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/jonny-yang/pen/ZEBLBzJ?editors=0010

#### Actual behavior:

Keyboard focus is moving on all the check boxes within Nav two times while navigating using down arrow key

#### Expected behavior:

Keyboard focus should move on all the check boxes only one time while navigating using down arrow key

### Documentation describing expected behavior

https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html#:~:text=Specific%20Benefits%20of%20Success%20Criterion,with%20the%20sequential%20reading%20order.&text=Such%20a%20user%20may%20interpret,focus%20order%20is%20not%20logical.

**Comment by joschect** [comment]
@yangster23  Thanks for the submission! I believe [this codepen](https://codepen.io/slimemold/pen/BaQRXYV?editors=0011) accomplishes the goal that you have. You'll need to control the click state yourself because focus never goes to the checkbox itself but only goes to the entire row. 

I think that Nav is intended behavior since the items are wrapped in a focus zone and anything that can be focused normally should be able to get focus.

**Comment by yangster23** [comment]
This doesn't address accessibility criteria though: when a keyboard user navigates to the nav bar, they'll get confused that there requires two "DOWN" key strokes to navigate to the next row. Given that the focus never goes to the checkbox itself, but only to the entire row when clicking, shouldn't keyboard navigation follow the same behavior - that focus only goes to the entire row?

**Comment by joschect** [comment]
@yangster23  I am a bit confused, the codepen that I sent you only requires 1 keypress to go down. The checkbox never gets focus. Is your goal to have the checkbox be a separate target from the entire row but not get selected on keyboarding?

**Comment by yangster23** [comment]
Apologies, I misread the codepen that you sent. The behavior in the codepen is correct; however, I get a typescript error where Type '{ "data-is-focusable": false; }' is not assignable to type 'ButtonHTMLAttributes<HTMLElement | HTMLButtonElement>'.
  Object literal may only specify known properties, and '"data-is-focusable"' does not exist in type 'ButtonHTMLAttributes<HTMLElement | HTMLButtonElement>'.  TS2322

    158 |           tabIndex={-1}
    159 |           inputProps={{
  > 160 |             "data-is-focusable": false,
        |             ^
    161 |           }}
    162 |         />
    163 |       </span>


Would you happen to know how to avoid this typescript error?

**Comment by ecraig12345** [comment]
In that case you unfortunately just have to cast the `inputProps` object to `any`, because prior to typescript 4.1 there's no way to specify that an interface allows all props starting with `data-`.
